app/controllers/admin/categories_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Admin
+  class CategoriesController < ApplicationController
+
+      def index
+        # show all
+        @categories = Category.all
+      end
+
+      def show
+        #shows one id
+        @category = Category.find(params[:id])
+      end
+
+      def new
+        @category = Category.new
+      end
+
+      def create
+        @category = Category.new(category_params)
+          if @category.save
+            redirect_to admin_categories_url
+          else
+            render 'new'
+          end
+      end
+
+      def edit
+        @category = Category.find(params[:id])
+      end
+
+      def update
+        @category = Category.find(params[:id])
+          if @category.update_attributes(category_params)
```

Comment:
@shaniajubelag Indent this block backward

Reply:
@inahsanjuan I have a different folder for categories for the users